pkg/services/sqlstore/migrations/usermig/test/user_lowercase_login_and_email_test.go

```diff
@@ -0,0 +1,136 @@
+package test
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	"github.com/grafana/grafana/pkg/infra/log"
+	"github.com/grafana/grafana/pkg/services/sqlstore/migrations/usermig"
+	"github.com/grafana/grafana/pkg/services/sqlstore/migrator"
+	"github.com/grafana/grafana/pkg/services/user"
+	"github.com/grafana/grafana/pkg/setting"
+)
+
+func TestLowerCaseMigration(t *testing.T) {
+	// Run initial migration to have a working DB
+	x := setupTestDB(t)
+
+	type migrationTestCase struct {
+		desc      string
+		users     []*user.User
+		wantUsers []*user.User
+	}
+	testCases := []migrationTestCase{
```

Comment:
missing the cases:
"2 users - same login, one already lowercase"
"2 users - same loging, none lowercase"